src/libMessage/Messenger.cpp

```diff
@@ -2227,6 +2248,27 @@ bool Messenger::SetNodeForwardTxnBlock(
     txnsGeneratedCount++;
   }
 
+  Signature signature;
+  if (result.transactions().size() > 0) {
+    // Serialize the transactions to temp buffer, later use it to calculate the
+    // signature.
+    std::vector<unsigned char> tmp;
+    int tempOffset = 0;
+    for (int i = 0; i < result.transactions().size(); ++i) {
+      tmp.resize(tmp.size() + result.transactions(i).ByteSize());
```

Comment:
SerializeToArray already does a resize if necessary. Please check the other parts of the code too for similar changes.